src/libraries/System.Numerics.Tensors/src/System/NativeIndex.cs

```diff
@@ -0,0 +1,162 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using System.Runtime.CompilerServices;
+
+namespace System
+{
+    /// <summary>Represent a type can be used to index a collection either from the start or the end.</summary>
+    /// <remarks>
+    /// <code>
+    /// int[] someArray = new int[5] { 1, 2, 3, 4, 5 } ;
+    /// int lastElement = someArray[^1]; // lastElement = 5
+    /// </code>
+    /// </remarks>
+    public readonly struct NativeIndex : IEquatable<NativeIndex>
+    {
+        private readonly nint _value;
+
+        /// <summary>Construct a NativeIndex using a value and indicating if the NativeIndex is from the start or from the end.</summary>
```

Comment:
There's a lot of places where it looks like `Index` was globally replaced with `NativeIndex` and that it included `index` (lowercase).

The documentation for System.Index is currently
> Represents a type that can be used to index a collection either from the beginning or the end.

and this should probably try to mirror that

Reply:
I'm sure thats what happened. I tried to catch where I did that but obviuosly missed some so good catch.

Reply:
Should be fixed now.